r/tests/testthat/test-dataset.R

```diff
@@ -453,15 +453,38 @@ test_that("Creating UnionDataset", {
 })
 
 test_that("map_batches", {
-  skip("map_batches() is broken (ARROW-14029)")
   ds <- open_dataset(dataset_dir, partitioning = "part")
+
+  # summarize returns arrow_dplyr_query
   expect_equal(
     ds %>%
       filter(int > 5) %>%
       select(int, lgl) %>%
-      map_batches(~ summarize(., min_int = min(int))),
+      map_batches(~ summarize(., min_int = min(int))) %>%
+      arrange(min_int),
     tibble(min_int = c(6L, 101L))
   )
+
+  # $num_rows returns integer vector
+  expect_equal(
+    ds %>%
+      filter(int > 5) %>%
+      select(int, lgl) %>%
+      map_batches(~ .$num_rows, .data.frame = FALSE) %>%
+      as.numeric() %>%
+      sort(),
+    c(5, 10)
```

Comment:
```suggestion
      map_batches(~ .$num_rows, .data.frame = FALSE) %>%
      sort(),
    c(5L, 10L)
```

Does this work? `as.numeric()` in there is a little suspicious — what issues do you have if you take it out?

Reply:
`map_batches()` will return a list, but `sort()` only takes atomic vectors.

Reply:
Aaaaaah, I see now. Maybe it would be a bit cleared with `unlist()` instead of `as.numeric()`?